website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -675,10 +746,31 @@ the types of the input and output elements. If your `DoFn` processes incoming
 look like this:
 {{< /paragraph >}}
 
+{{< paragraph class="language-go" >}}
+A `DoFn` processes one element at a time from the input `PCollection`. When you
```

Comment:
Suggested change:

A `DoFn` processes one element at a time from the input `PCollection`. When you

to

A `DoFn` processes one element at a time from the input `PCollection`. A `DoFn` can be either a `struct` or a `func`. When you

Reply:
I'm going to push back on this at present. 

The lack of func nature is apparent in the example, immediately adjacent to the text, and shouldn't be over explained right here. At least, not at this point in the document. There's an entire section on syntactic sugars and convenience abbreviations for DoFns, which introduces that funcs can be DoFns too. I've introduced functional DoFns there since they are largely conveniences, vs the Structural DoFns.